mobile/native/src/ln_dlc/node.rs

```diff
@@ -155,89 +157,90 @@ impl Node {
-                                }
-                                Err(e) => {
-                                    tracing::warn!(
-                                        "Could not find a filling position in the database.
-                                         Maybe because the coordinator closed an expired position.
-                                         Error: {e:#}"
-                                    );
-
-                                    tokio::spawn(async {
-                                        match position::handler::close_position().await {
-                                                Ok(_) => tracing::info!("Successfully closed expired position."),
-                                                Err(e) => tracing::error!("Critical Error! We have a DLC but were unable to set the order to filled. Error: {e:?}")
-                                            }
-                                    });
-                                }
-                            };
-                        }
-                        _ => (),
-                    }
-                }
+                resp
             }
+        };
+
+        if let Some(msg) = resp {
+            self.send_dlc_message(node_id, msg)?;
         }
 
         Ok(())
     }
 
+    pub fn send_dlc_message(&self, node_id: PublicKey, msg: Message) -> Result<()> {
+        tracing::info!(
+            to = %node_id,
+            kind = %dlc_message_name(&msg),
+            "Sending message"
+        );
+
+        self.inner
+            .dlc_message_handler
+            .send_message(node_id, msg.clone());
+
+        // After sending certain messages, we need to do some post-processing
+        match msg {
+            Message::SubChannel(SubChannelMessage::Finalize(SubChannelFinalize {
+                channel_id,
+                ..
+            })) => {
+                let contracts = self.inner.dlc_manager.get_store().get_contracts()?;
+
+                let accept_collateral = contracts
+                    .iter()
+                    // Taking the first `Confirmed` contract we find is just a
+                    // heuristic. Ideally we would be able to match against the
+                    // `ContractId` or the `ChannelId`, but the information is not
+                    // guaranteed to be there
+                    .find_map(|contract| match contract {
+                        Contract::Confirmed(SignedContract {
+                            accepted_contract, ..
+                        }) => Some(accepted_contract.accept_params.collateral),
+                        _ => None,
+                    })
+                    .with_context(|| {
+                        format!(
+                            "Confirmed contract not found for channel ID: {}",
+                            hex::encode(channel_id)
+                        )
+                    })?;
+
+                let filled_order = order::handler::order_filled()
```

Comment:
It's a really weird situation because the DLC channel has been finalised, so there is no going back.

The modelling is not perfect here because a finalised DLC necessarily implies that an order was filled and yet we can end up with inconsistent states.

I think for the order it is somewhat less important as it's mostly part of the history, but if we don't update the position accordingly the user doesn't know what's going on at all. My intuition is that this should be fixed with modelling: the position should just be a reflection of the state from `rust-dlc` (and `ln-dlc-node`) and we should only commit to the database "metadata" and no state.